packages/amino/src/secp256k1hdwallet.ts

```diff
@@ -107,6 +107,12 @@ interface DerivationInfo {
   readonly prefix: string;
 }
 
+export interface Secp256k1HdWalletOptions {
+  readonly bip39Password?: string;
+  readonly hdPath?: HdPath;
+  readonly prefix?: string;
+}
```

Comment:
What about making the fields required and using `Partial<Secp256k1HdWalletOptions>`? Then we can use `Secp256k1HdWalletOptions` for places where all options are set (e.g. after replacing unset with defaults).

Reply:
Updated. I'm not sure how you wanted to handle the `bip39Password` option though: I left it optional, but maybe you prefer the type to be `string | undefined`, or we could even give it an empty string as default since it's just appended to the salt.

Reply:
I would default to an empty string because this is how the algorithm is defined:

> To create a binary seed from the mnemonic, we use the PBKDF2 function with a mnemonic sentence (in UTF-8 NFKD) used as the password and the string "mnemonic" + passphrase (again in UTF-8 NFKD) used as the salt.

https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki#From_mnemonic_to_seed

For the implementation `undefined` and `""` are synonyms.